crates/via-router/src/path/pattern.rs

```diff
@@ -12,33 +13,16 @@ pub enum Pattern {
 ///
 #[derive(Debug, PartialEq)]
 pub struct Param {
-    ident: Box<str>,
-}
-
-/// Unwraps the remaining slice of a path segment after the first char or
-/// panics with a custom message.
-///
-macro_rules! rest_or {
-    (
-        // Should evaluate to a `&str`.
-        $segment:expr,
-        // Args passed to panic! if the segment is empty.
-        $($arg:tt)*
-    ) => {{
-        let segment = $segment;
-        match segment.get(1..) {
-            // The range is out of bounds or produced an empty str.
-            None | Some("") => panic!($($arg)*),
-            // The range is valid.
-            Some(rest) => rest,
-        }
-    }};
+    ident: Arc<str>,
```

Comment:
Didn't mean to this here! I was experimenting with using `Arc` for params again.